src/Symfony/Component/Security/Http/HttpUtils.php

```diff
@@ -135,6 +135,10 @@ public function generateUri($request, $path)
             return $path;
         }
 
+        if (!$request instanceof Request) {
```

Comment:
Fair enough :)) I guess the thing is we want to improve method signatures / type hints in a upcoming major release, but the policy makes it real hard. And personally i feel it's holding us back from improving (read: simplifying) the codebase.

Ie. this is _not_ about adding safety guards, scalar type checks, etc. but about forcing integrity where possible.

Anyway, as any error will be clear enough probably i understand it's not worth the effort.